_imaging.c

```diff
@@ -1905,6 +1905,32 @@ _putband(ImagingObject* self, PyObject* args)
     return Py_None;
 }
 
+static PyObject*
+_split(ImagingObject* self, PyObject* args)
+{
+    int fails = 0;
+    Py_ssize_t i;
+    PyObject* list;
+    PyObject* imaging_object;
+    Imaging bands[4] = {NULL, NULL, NULL, NULL};
+
+    if ( ! ImagingSplit(self->image, bands))
+        return NULL;
+
+    list = PyTuple_New(self->image->bands);
+    for (i = 0; i < self->image->bands; i++) {
+        imaging_object = PyImagingNew(bands[i]);
+        if ( ! imaging_object)
+            fails += 1;
+        PyTuple_SET_ITEM(list, i, imaging_object);
+    }
+    if (fails) {
+        Py_DECREF(list);
```

Comment:
`PyImagingNew` will call `ImagingDelete` for failed `Imaging` objects. Other `Imaging` objects are wrapped in `ImagingObject` objects and stored in `list`. That is why when we dereferencing `list`, all stored images are also freeing.